src/storage.py

```diff
@@ -150,6 +219,8 @@ def remove_osd(self, osd_num: int, force: bool = False):
             microceph.remove_disk_cmd(osd_num, force)
             # if no errors while removing OSD, clean stale osd records.
             self._clean_stale_osd_data()
+            # Request charm to update all clients.
+            self.on.unit_storage_changed.emit()
         except CalledProcessError as e:
             if force:
```

Comment:
shouldn't we notify of changes to clients here as well?

Reply:
ack, nice catch. have added.